scripts/jenkins/cloud/ansible/roles/ssh_keys/tasks/main.yml

```diff
@@ -25,3 +25,15 @@
       {{ key }}
       {% endfor %}
       {% endfor %}
+
+- name: Create local SSH key on deployer
+  shell: ssh-keygen -t rsa -f "{{ ansible_env.HOME }}/.ssh/id_rsa" -N ""
+  delegate_to: localhost
+  args:
+    creates: "{{ ansible_env.HOME }}/.ssh/id_rsa"
+
+- name: Copy local SSH key to deployer
+  lineinfile:
+    path: "{{ ansible_env.HOME }}/.ssh/authorized_keys"
+    create: yes
+    line: "{{ lookup('file', ansible_env.HOME + '/.ssh/id_rsa.pub') }}"
```

Comment:
With this one can't not use this key (when another one is already used). You could edit https://github.com/SUSE-Cloud/automation/blob/master/scripts/jenkins/cloud/ansible/group_vars/all/ssh_pub_keys.yml localy, which is used in the code above these additions.

Reply:
I think this is fine, this task will append the public key to `authorized_keys` and will only create a file if it does not already exists. Considering that the previous task already has `create: yes` you could even remove this parameter from this task.

This would also remove this requirement: https://github.com/SUSE-Cloud/automation/blob/master/docs/ardana/testing.md#L191

Reply:
@JanZerebecki I don't like to edit files locally that are under version control unless I really mean to commit those changes later. And the key I want to add might have a very show lifetime, see #3497 .